drivers/serial/Kconfig.nrfx

```diff
@@ -69,6 +76,17 @@ config UART_0_NRF_TX_BUFFER_SIZE
 	  This value is limited by range of TXD.MAXCNT register for
 	  particular SoC.
 
+config UART_0_NRF_HW_ASYNC
+	bool "Use hardware RX byte counting"
+	depends on UART_0_NRF_UARTE
+	depends on UART_ASYNC_API
+	help
+	  Hardware RX byte counting requires timer instance and one PPI channel
+
+config UART_0_NRF_HW_ASYNC_TIMER
+	int "Timer instance"
```

Comment:
There is no default value, user has to set it explicitly

Reply:
So as for now default value is `n`

Reply:
If I understand kconfig properly (@ulfalizer can help) this will be a bit undefined, i.e. it's not really a "not-set", cause the value is boolean. So, it appears to me that the macro is still generated (Without a value though)

Reply:
If value is not set then cmake will report an error and won't generate the project

Reply:
We should give the developer the option to build this without menuconfig, and as far as I can understand Kconfig, this requires a default value for int- Kconfig symbols.

Reply:
This value is taken into account only when developer selects hardware byte counting. When build with default configuration there are no issues with it. I don't want to select a default value here, as it should be explicitly set by developer if he decides to use hardware byte counting.

Reply:
Understood, now, thanks. So, I suggest you do as @ulfalizer has done in several similar cases, i.e. remove the "depends on" and add the whole Kconfig definition in i`f UART_0_NRF_HW_ASYNC ... endif #UART_0_NRF_HW_ASYNC`

Reply:
There's some notes on `int` defaults on the [best practices page](https://docs.zephyrproject.org/latest/guides/kconfig/index.html#redundant-defaults).

Usually, it's a good idea to give `int` symbols defaults, because they don't default to zero (for semi-messy reasons). If no default is given, then the `prj.conf` (or board `defconfig` files, etc.) must set the symbol. Otherwise, you'll get a `.config` that generates warnings when read back in.

(Super nit: Trying to standardize making it `# UART_0_NRF_HW_ASYNC` instead of `#UART_0_NRF_HW_ASYNC` too btw. It's more common.)

Reply:
Re. `if` vs `depends on`, `depends on` is cleaner if there's just a single symbol with the dependency.

If there's many symbols with the same dependency in a row, `if` is cleaner.

It's a bit subjective when you have just two symbols or the like. It's good to be aware that `if FOO` is just a shorthand for adding `depends on FOO` to all the stuff within at least.

People tend to read a bunch of things into Kconfig that aren't there (probably because the official docs aren't great). :)